Cabal/Distribution/Simple/Utils/Json.hs

```diff
@@ -0,0 +1,34 @@
+module Distribution.Simple.Utils.Json
+    ( Json(..)
+    , renderJson
+    ) where
+
+data Json = JsonArray [Json]
+          | JsonBool !Bool
+          | JsonNull
+          | JsonNumber !Int
+          | JsonObject [(String, Json)]
+          | JsonString !String
+
+renderJson :: Json -> ShowS
+renderJson (JsonArray objs)   =
+  surround "[" "]" $ intercalate "," $ map renderJson objs
+renderJson (JsonBool True)    = showString "true"
+renderJson (JsonBool False)   = showString "false"
+renderJson  JsonNull          = showString "null"
+renderJson (JsonNumber n)     = shows n
+renderJson (JsonObject attrs) =
+  surround "{" "}" $ intercalate "," $ map render attrs
+  where
+    render (k,v) = (surround "\"" "\"" $ showString k) . showString ":" . renderJson v
+renderJson (JsonString s)     = surround "\"" "\"" $ showString s
```

Comment:
I think this doesn't quote properly. When trying this out on `exe:cabal` I get:

```
{"cabal_version":"3.0.0.0","compiler":{"flavour":"GHC","compiler_id":"ghc-8.4.4","path":"/usr/bin/ghc"},"components":[{"type":"executable","name":"CExeName (UnqualComponentName "cabal")", [...]
```

The `"CExeName (UnqualComponentName "cabal")"` bit is obviously not proper json.

Hoenstly you should just be using a proper pretty printer for the component name and not just `show` anyways.

Reply:
I observed the same behaviour, will try to fix

Reply:
Do we have a fitting printing lib? Couldn't find one. I could also try to migrate to `microaeson`, but this would introduce a new dependency.

Reply:
I think adding a dependency is not really necessary. We just have to make sure the string doesn't contain any quotes. I don't remember which function does that but there is a pretty printer for these that gives you the format cabal uses on the command-line, like `exe:cabal` or `lib:Cabal`. That shouldn't have this problem in the first place.

Reply:
Should be fixed now!